### PR TITLE
feat(specification-validation) implement Non-Repeatable Required 1XX Fields validation

### PR DESCRIPTION
**Pull request by viacheslavpoliakov**
### Purpose
Implement Non-Repeatable Required 1XX Fields validation

### Approach
Add NonRepeatableRequired1XXFields validator

### Changes Checklist
- [ ] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [ ] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [ ] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MRSPECS-42](https://folio-org.atlassian.net/browse/MRSPECS-42)

### CODE REVIEW

File: mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatableRequired1xxFieldRuleValidator.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  extends Abstract1xxFieldRuleValidator {
+
+  MarcFieldNonRepeatableRequired1xxFieldRuleValidator(TranslationProvider translationProvider) {
+    super(translationProvider);
+  }
+
+  @Override
+  public List<ValidationError> validate(Map<String, List<MarcField>> fields, SpecificationDto specification) {
+    var errors = new ArrayList<ValidationError>();
+    if (fields.keySet().stream().noneMatch(TagsMatcher::matches1xx)) {
```
Comment (viacheslavkol) [inline mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatableRequired1xxFieldRuleValidator.java:24]: Shouldn't this validation return an error also if there are multiple 1XX?

Comment (viacheslavpoliakov) [reply mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatableRequired1xxFieldRuleValidator.java:24]: Yep, will add it after MRSPECS-41 fixing

Comment (viacheslavpoliakov) [reply mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatableRequired1xxFieldRuleValidator.java]: Fixed

File: mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatable1xxFieldRuleValidator.java [hunk 2]
```diff
@@ -2,33 +2,27 @@
...
   @Override
   public List<ValidationError> validate(Map<String, List<MarcField>> fields, SpecificationDto specification) {
-    List<MarcField> all1xxFields = fields.keySet().stream()
-      .filter(TagsMatcher::matches1xx)
-      .flatMap(tag -> fields.get(tag).stream())
-      .toList();
-
+    if (!FamilyProfile.BIBLIOGRAPHIC.equals(specification.getProfile())) {
+      return List.of();
+    }
```
Comment (psmagin) [inline mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatable1xxFieldRuleValidator.java:24]: Please do not rely on family profile. Rules are not specific to any profile.

**Review by psmagin — CHANGES_REQUESTED** [review]

File: mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatable1xxFieldRuleValidator.java (hunk 2 above)
Comment (viacheslavpoliakov) [reply mod-record-specifications-validator/src/main/java/org/folio/rspec/validation/validator/marc/impl/MarcFieldNonRepeatable1xxFieldRuleValidator.java]: Oh yeah, fixed.

**Review by psmagin — APPROVED** [review]

**Review by viacheslavkol — APPROVED** [review]

**Pull request merged**